src/main/scala/io/iohk/ethereum/ledger/BlockExecution.scala

```diff
@@ -67,7 +67,13 @@ class BlockExecution(
     * @param block the block with transactions to run
     */
   private[ledger] def executeBlockTransactions(block: Block): Either[BlockExecutionError, BlockResult] = {
```

Comment:
maybe it would be possible to pass parent header here as param , to avoid this `blockchain.getBlockHeaderByHash(block.header.parentHash` ?